### DISCUSSION TITLE
bind-addr ignored

### DISCUSSION THREAD

**Discussion by Aetherinox** [coder/code-server#6859] [Q&A]
I decided to post a discussion instead of issue, because I hope this is something dumb on my end.
No matter how I configure it, it insists on using port **8080**.

Edited the `config.yaml`
```yml
bind-addr: 127.0.0.1:8443
cert: true
auth: password
```

```
info  code-server 4.90.3 26c763485b9101fcae2ce56fc0903f0e2d48abe3
info  Using user-data-dir /home/coder/.local/share/code-server
info  Using config file /home/coder/.config/code-server/config.yaml
info  HTTPS server listening on https://0.0.0.0:8080/
info    - Authentication is enabled
info      - Using password from /home/coder/.config/code-server/config.yaml
info    - Using certificate for HTTPS: /home/coder/.local/share/code-server/localhost.crt
info  Session server listening on /home/coder/.local/share/code-server/code-server-ipc.sock
```

I've tried putting `cert` first
Tried `bind-addr:` first to see if it was a sorting issue
Tried `bind-addr: 0.0.0.0:8443`
Tried `cert: false`

No matter what, it refuses to change the port. The only way I can force the port to change is by adding:

```yml
    container_name: vscode
    image: codercom/code-server
    entrypoint: /usr/bin/entrypoint.sh --bind-addr 0.0.0.0:8443
```

I have to add an entrypoint. But then I lose `cert` unless I add it there.

The weird thing is that it is indeed reading the config, because it's seeing `cert: true`, and the password is correct.

I enabled debugging just to see if anything extra would come up, and I saw:
```
debug parsed command line {"args":{"bind-addr":"0.0.0.0:8080","_":["."]}}
debug parsed config {"args":{"bind-addr":"127.0.0.1:8443","cert":{},"auth":"password"}}
debug parent:33 spawned child process 56
debug child:56 initiating handshake
debug parent:33 got message {"message":{"type":"handshake"}}
debug child:56 got message  {"message":{"type":"handshake","args":{"bind-addr":"0.0.0.0:8080","cert":{"value":"/home/coder/.local/share/code-server/localhost.crt"},"auth":"password","config":"/home/coder/.config/code-server/config.yaml","_":["/home/coder"],"user-data-dir":"/home/coder/.local/share/code-server","extensions-dir":"/home/coder/.local/share/code-server/extensions","session-socket":"/home/coder/.local/share/code-server/code-server-ipc.sock","log":"trace","verbose":true,"host":"0.0.0.0","port":8080,"cert-key":"/home/coder/.local/share/code-server/localhost.key","proxy-domain":[],"usingEnvPassword":false,"usingEnvHashedPassword":false}}}
```

It appears to be parsing command-line first, and I have no args or env variables defined in my compose file for an IP or port, it's strictly just in the config.yaml

Only way I got it up and operational was

```yml
entrypoint: /usr/bin/entrypoint.sh --bind-addr 0.0.0.0:8443 --cert
```

Yet my argon hashed password in the config.yml is read fine.

**Comment by code-asher** [comment]
Ah yeah the command line flags take precedence over the config file, and we have `--bind-addr` set in the entrypoint:

https://github.com/coder/code-server/blob/c7d88abf9f57d0576f913b9365532f0e2b885c0c/ci/release-image/Dockerfile#L58

So, adding your own `--bind-addr` is indeed the only way to change the address with the Docker image.  You can override the entrypoint like you are doing here, or I think you could also add `--bind-addr` as part of the command (if a flag appears twice the second should take precedence).

**Reply by code-asher** [reply]
I think maybe we should change our entrypoint to `--host 0.0.0.0` instead of using `--bind-addr`; the `host` flag is technically deprecated, but that would at least make it possible to change the port with the config file.

**Reply by Aetherinox** [reply]
That would be a good solution. I sat here for hours wondering what I did wrong and tried about every combination I could find to get it to work, but eventually I had to use an entrypoint, even though all my other settings are in a config.

Or maybe set it to prioritizing the 2nd of the two. Whichever seems more logical. Because if I add one to the config, then obviously that's the one I want to use, otherwise it defaults to whatever is specified in the entrypoint.
